src/java_bytecode/java_bytecode_convert_method.cpp

```diff
@@ -2722,9 +2725,8 @@ const bool java_bytecode_convert_methodt::is_method_inherited(
         return true;
       // methods with the default access modifier are only
       // accessible within the same package.
-      else if(access==ID_default &&
-              java_class_to_package(id2string(parent))==classpackage)
-        return true;
+      else if(access==ID_default)
+        return java_class_to_package(id2string(parent))==classpackage;
```

Comment:
I don't think so. Note that I wrote "When we find a class in a different package".

Anyway, the function got almost entirely rewritten and there is no loop anymore.